vowpalwabbit/io_to_queue.h

```diff
@@ -0,0 +1,76 @@
+#ifndef _IO_QUEUE_
+#define _IO_QUEUE_
+
+#include <thread>
+#include <future>
+#include <mutex>  
+#include <chrono>   
+
+#include "io_item.h"
+
+//Adds a line of input to the input queue
+inline bool add_to_queue(vw& all, char *& line){
```

Comment:
Most of these should be moved into IO_State (more details in the comments there)